classes/checkout/CheckoutAddressesStep.php

```diff
@@ -202,18 +202,40 @@ public function handleRequest(array $requestParams = array())
 
     public function getTemplateParameters()
     {
+        $idAddressDelivery = (int) $this->getCheckoutSession()->getIdAddressDelivery();
+        $idAddressInvoice = (int) $this->getCheckoutSession()->getIdAddressInvoice();
+        $addressWarning = false;
+        $errorKey = 'address_error';
+        $checkoutWarning = isset($this->context->controller->checkoutWarning['address']) ? $this->context->controller->checkoutWarning['address'] : false;
```

Comment:
I'm not a big fan of this, but I'm struggling to find another, cleaner way to allow the controller to pass this information along